collectors/google_pubsub/pubsub_collector.go

```diff
@@ -36,19 +60,79 @@ func (collector *PubSubCollector) HandleMessage(message *pubsub.Message) error {
 	return result.Err
 }
 
+func (collector *PubSubCollector) PollForMessages() {
+	// QUESTION - should I add the channel usage back in here like in AWS Collector?
+	for {
+		if !collector.BaseCollector.Sleeping {
+			fmt.Println("Processing messages...")
+			messages, err := collector.GetMessages()
+			if err != nil {
+				fmt.Println("Error recieving messages - retrying")
+				collector.ProcessExponentialBackoff(err)
+			}
+			for _, msg := range messages {
+				collector.BusinessProcessor(msg.MessageBody)
+			}
+			fmt.Printf("Recieved messages - %s\n", messages)
+		}
+	}
+}
+
 func (collector *PubSubCollector) GetMessages() ([]collectors.MessageWrapper, error) {
-	var err error
+	// There's a built in deadline on the subscription that will orphan the message in the subscription.
+	subscription := collector.PubSubClient.Subscription(collector.SourceSubscription)
 	messages := make([]collectors.MessageWrapper, 0)
 
+	// NOTE - HTF is this referencing the subscription.Mutex?!?!?!?!
+	var mu sync.Mutex
+	err := subscription.Receive(collector.Context, func(ctx context.Context, msg *pubsub.Message) {
```

Comment:
You can use this for now:  If we want a timeout on the context we can do that later.
```suggestion
	err := subscription.Receive(context.Background(), func(ctx context.Context, msg *pubsub.Message) {
```

Reply:
Nice!  I used this and added the cancel functionality to it.